### DISCUSSION TITLE
Sorting items in a collection

### DISCUSSION THREAD

**Discussion by schmt5** [Thinkmill/keystatic#406] [Q&A]
Is there a way to sort items in a collections? Either inside the keystatic-Admin Panel or by the `Reader API`?

**Comment by JedWatson** [comment]
See my answer here for the full background, sorting is very similar to filtering: https://github.com/Thinkmill/keystatic/discussions/357#discussioncomment-6304117

There are two parts to this, I'll answer them separately

## What you can do now

You can't specifically sort entries (what you call items, we call entries) in a collection in Keystatic, they're ordered by the slug.

You _can_ generate the slug in the way you want the entries ordered, e.g by putting a timestamp at the start, etc.

I've also seen people add an `order` field to the schema and store an integer in the front-matter that can be used to sort the entries when they're read, but this isn't easy to manage in the Admin UI yet.

To sort entries when you read them with the Reader API, just use JavaScript, e.g if you wanted them sorted by date (and have a date field in the entry schema):

```
import { compareAsc } from 'date-fns'

const posts = await reader.collections.blog.all();
posts.sort(compareAsc);
```

(array sort docs [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort))

## What we have planned

Like I mention in #357 and #375 we **do** plan to add columns to the Admin UI, and make entries sortable, which will improve usability a lot in this case.

We're also exploring adding drag and drop functionality to the collection view to make it easier to reorder entries (e.g if you're using an `order` number field) but I can't say when this will land.

I expect the Reader API will stay as it is, because we'd prefer not to prescribe what you can do with it, and encourage just using JavaScript instead (like in the example above, you could use `date-fns` or any other library you prefer to sort the entries by date)

**Reply by schmt5** [reply]
Thanks for your detailed answer. 

I didn't know, that entries in a collection get ordered by the slug. I guess it's a normal _JS-String-Compare_. So at the moment it can be done with slugs like` a-myFirstEntry`, `b-mySecondEntry` (I gonna try it). This requires knowledge from the content creators, but I think that's fine at the moment.

I also came up with the idea of creating a `singleton` with a `relationship` field to sort a collection.
